sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/service/cosmos/CosmosClientBuilderFactory.java

```diff
@@ -76,16 +98,55 @@ protected void configureService(CosmosClientBuilder builder) {
         map.from(this.cosmosProperties.getReadRequestsFallbackEnabled()).to(builder::readRequestsFallbackEnabled);
         map.from(this.cosmosProperties.getSessionCapturingOverrideEnabled()).to(builder::sessionCapturingOverrideEnabled);
         map.from(this.cosmosProperties.getPreferredRegions()).whenNot(List::isEmpty).to(builder::preferredRegions);
-        map.from(this.cosmosProperties.getThrottlingRetryOptions()).to(builder::throttlingRetryOptions);
+
+        ThrottlingRetryOptions retryOptions = this.cosmosProperties.getThrottlingRetryOptions();
+        if (this.throttlingRetryOptions != null && isDefaultThrottlingRetryOptions(retryOptions)) {
+            map.from(this.throttlingRetryOptions).to(builder::throttlingRetryOptions);
+            LOGGER.debug("The throttling retry options is not configured, "
+                + "then the Azure Spring Retry configuration will be applied to Cosmos service builder.");
+        } else {
+            map.from(retryOptions).to(builder::throttlingRetryOptions);
+        }
 
         // TODO (xiada): should we count this as authentication
         map.from(this.cosmosProperties.getResourceToken()).to(builder::resourceToken);
         map.from(this.cosmosProperties.getPermissions()).whenNot(List::isEmpty).to(builder::permissions);
+        GatewayConnectionConfig gatewayConnection = this.cosmosProperties.getGatewayConnection();
+        if (proxyOptions != null && gatewayConnection.getProxy() == null) {
+            gatewayConnection.setProxy(proxyOptions);
+            LOGGER.debug("The proxy of the Gateway connection is not configured, "
+                + "then the Azure Spring Proxy configuration will be applied to Cosmos gateway connection.");
+        }
         if (ConnectionMode.DIRECT.equals(this.cosmosProperties.getConnectionMode())) {
-            builder.directMode(this.cosmosProperties.getDirectConnection(), this.cosmosProperties.getGatewayConnection());
+            builder.directMode(this.cosmosProperties.getDirectConnection(), gatewayConnection);
         } else if (ConnectionMode.GATEWAY.equals(this.cosmosProperties.getConnectionMode())) {
-            builder.gatewayMode(this.cosmosProperties.getGatewayConnection());
+            builder.gatewayMode(gatewayConnection);
+        }
+    }
+
+    /**
+     * Check if the retry option is the default value, which is defined in azure-cosmos SDK.
+     * @param retryOptions retry options to be checked
+     * @return result
+     */
+    private boolean isDefaultThrottlingRetryOptions(ThrottlingRetryOptions retryOptions) {
+        if (retryOptions.getMaxRetryWaitTime().equals(Duration.ofSeconds(30))
+            && retryOptions.getMaxRetryAttemptsOnThrottledRequests() == 9) {
+            return true;
+        }
```

Comment:
This is hard code, and this will couple our logic with the SDK implementation. When the SDK changes the default configuration, this assertion won't be valid anymore.